### PR TITLE
Fixed defines and added `cgl_sharegroup` support

### PR DESCRIPTION
**Pull request by Kerosene2000**

### CODE REVIEW

File: src/core/mod.rs [hunk 1]
```diff
@@ -646,7 +652,7 @@ enum_from_primitive! {
         GlContextKhr = ffi::CL_GL_CONTEXT_KHR as isize,
         EglDisplayKhr = ffi::CL_EGL_DISPLAY_KHR as isize,
         GlxDisplayKhr = ffi::CL_GLX_DISPLAY_KHR as isize,
-        CglSharegroupKhr = ffi::CL_CGL_SHAREGROUP_KHR as isize,
+        CglSharegroupKhr = CL_CGL_SHAREGROUP_KHR_OS_SPECIFIC as isize, // FIXME that's ugly
```
Comment (Kerosene2000) [inline src/core/mod.rs]: This is ugly. How can we do ?

File: src/core/mod.rs [hunk 2]
```diff
@@ -635,6 +635,12 @@ enum_from_primitive! {
     }
 }
 
+// FIXME that's ugly
+#[cfg(target_os = "macos")]
+const CL_CGL_SHAREGROUP_KHR_OS_SPECIFIC: u32 = ffi::CL_CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE;
+#[cfg(not(target_os = "macos"))]
+const CL_CGL_SHAREGROUP_KHR_OS_SPECIFIC: u32 = ffi::CL_CGL_SHAREGROUP_KHR;
+
```
Comment (Kerosene2000) [inline src/core/mod.rs]: It goes together...

**Comment by Kerosene2000** [comment]
I have a working program (OS X tested), [particle-system-rs (rev = "b6edf6a")](https://github.com/Kerosene2000/particle-system-rs).
- Acquiring a `gl_buffer` (`vertexBuffer`)
- Reading all its values
- Adding `0.2` to each values (executing a kernel on it)
- Re-reading all its values
- Releasing it
- Displaying it with `glium`

File: src/core/mod.rs (hunk 1 above)
Comment (c0gent) [inline src/core/mod.rs]: Leave it ugly for now. We can rename it later if necessary without too much trouble.

**Comment by c0gent** [comment]
Glad to see you had a chance to get this working. I'll look at it as soon as I can.

**Comment by c0gent** [comment]
This will require some time for me to get working on Linux and Windows.

May have time this week.

**Comment by c0gent** [comment]
Please change the type of `CL_CGL_SHAREGROUP_KHR_OS_SPECIFIC` by duplicating the changes made in [this commit](https://github.com/cogciprocate/ocl/commit/dbea2117ce4314f8d4dfc92c729f95fcdd6458bd).

Test that on your system. If it works, I will merge.

**Comment by Kerosene2000** [comment]
That works fine on `OS X (10.11.5)` 👍

**Pull request merged**